ext/openssl/ossl_ssl.c

```diff
@@ -2081,19 +2083,27 @@ ossl_ssl_write_internal(VALUE self, VALUE str, VALUE opts)
         if (!NIL_P(cb_state)) {
             rb_ivar_set(self, ID_callback_state, Qnil);
             ossl_clear_error();
+            rb_str_unlocktmp(tmp);
             rb_jump_tag(NUM2INT(cb_state));
         }
 
         switch (ssl_get_error(ssl, nwritten)) {
           case SSL_ERROR_NONE:
+            rb_str_unlocktmp(tmp);
             return INT2NUM(nwritten);
           case SSL_ERROR_WANT_WRITE:
-            if (no_exception_p(opts)) { return sym_wait_writable; }
+            if (no_exception_p(opts)) {
+                rb_str_unlocktmp(tmp);
+                return sym_wait_writable;
+            }
             write_would_block(nonblock);
             io_wait_writable(io);
```

Comment:
`write_would_block()` and `io_wait_writable()` can raise an exception.

Reply:
I noticed that `#sysread` actually has this bug. I will fix it in another PR.

Reply:
Indeed. I wanted to avoid it, but I suppose that using `rb_protect` may be the only wait to properly catch all possible exits.